src/v/storage/tests/compaction_e2e_test.cc

```diff
@@ -1182,10 +1182,6 @@ TEST_P(
                     tombstone_retention_ms)
                     .get();
 
-    // Compaction will only have occurred if tombstones were eligible for
-    // deletion.
-    ASSERT_EQ(did_compact, wait_for_retention_ms);
-
```

Comment:
I've been leaning on `RPTEST_REQUIRE_EVENTUALLY` for stuff like this lately, but if you say it's redundant I'll take your word for it.